src/commons/listener.rs

```diff
@@ -103,6 +133,56 @@ pub struct ListenerStatus {
 pub struct ListenerIngress {
     /// The hostname or IP address to the [`Listener`].
     pub address: String,
+    /// The type of address (`Hostname` or `IP`).
+    pub address_type: AddressType,
     /// Port mapping table.
     pub ports: BTreeMap<String, i32>,
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug, JsonSchema)]
```

Comment:
The traits `Copy`, `Eq`, and `PartialEq` could be added to the (fieldless) enums and `Eq` and `PartialEq` to the structs. This saves us from making a new operator-rs release if the traits are needed in the future.

Reply:
Sure.

Reply:
df72972cd643851b85fb301f3e0da7641dbdd160